core/commands/ls.go

```diff
@@ -79,21 +87,42 @@ format:
 			}
 			for j, link := range dagnode.Links {
 				var linkNode *merkledag.Node
-				linkNode, err = link.GetNode(req.Context(), node.DAG)
-				if err != nil {
-					res.SetError(err, cmds.ErrNormal)
-					return
+				t := unixfspb.Data_DataType(-1)
+				linkKey := key.Key(link.Hash)
+				if ok, err := node.Blockstore.Has(linkKey); ok && err == nil {
+					b, err := node.Blockstore.Get(linkKey)
+					if err != nil {
+						res.SetError(err, cmds.ErrNormal)
+						return
+					}
+					linkNode, err = merkledag.DecodeProtobuf(b.Data())
```

Comment:
ok wat-- all this merkledag code shouldnt have to be out here. this should all be handled by util functions in either `unixfs` or some other package. this is leaking abstractions out